azurerm/internal/services/signalr/signalr_service_resource.go

```diff
@@ -390,6 +465,57 @@ func flattenSignalRFeatures(features *[]signalr.Feature) []interface{} {
 	return result
 }
 
+func expandUpstreamSettings(input []interface{}) *signalr.ServerlessUpstreamSettings {
+	upstreamTemplates := make([]signalr.UpstreamTemplate, 0)
+
+	for _, upstreamSetting := range input {
+		setting := upstreamSetting.(map[string]interface{})
+
+		upstreamTemplate := &signalr.UpstreamTemplate{
```

Comment:
there's no need for this to be a pointer:

```suggestion
		upstreamTemplate := signalr.UpstreamTemplate{
```